docs/design/storagenode-graceful-exit.md

```diff
@@ -91,6 +92,133 @@ This process including the Storage nodes transferring their pieces to other node
 	- Receiving storage nodes should be uploaded to as if they were receiving any other upload (orders, etc).
 - We need to have the storage node keep track how its progress during graceful exit by calculating how much data it had when it started the graceful exit and keeping track of how much data it has deleted (data is deleted from the exiting node as it sends pieces to new nodes). This is an estimation since the storage node is likely holding more data than it will gracefully exit (garbage data). 
 
+## Implementation (MVP)
+
+#### Satellite
+- Update DBX - Add updateable `exit_initiated_dt` and `exit_completed_dt` timestamps to nodes table with indexes
+  - Add GetExitingNodeIds method to overlaycache. Returns nodes IDs where `exit_initiated_dt` is not null and `exit_completed_dt` is null.
+- Create GracefulExit endpoint
+  - Endpoints should be secured using the peer Identity provided in context
+  - Initiates the exit by setting `nodes.exit_initiated` to current time
+  - ``` 
+	service GracefulExit {
+		rpc Initiate(stream InitiateRequest) returns (stream InitiateResponse) {}
+	}
+
+	message InitiateRequest {
```

Comment:
This is not a comment, but a thought... when I first read 'initiaterequest' didn't think it was referring to graceful exit , so what do you think, of naming it 'GracefulExitInitiateRequest'

Reply:
Updated to GracefulExit rpc InitiateExit(InitiateExitRequest) returns (InitiateExitResponse)